test/cluster_helper.go

```diff
@@ -381,6 +392,45 @@ func rollingRestartClusterByRemovingNamespaceDynamicallyTest(
 	return updateCluster(k8sClient, ctx, aeroCluster)
 }
 
+func validateServiceUpdate(k8sClient client.Client, ctx goctx.Context,
+	clusterNamespacedName types.NamespacedName, ports []int32) error {
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	serviceNamespacesNames := make([]types.NamespacedName, 0)
+	for podName := range aeroCluster.Status.Pods {
+		serviceNamespacesNames = append(serviceNamespacesNames,
+			types.NamespacedName{Name: podName, Namespace: clusterNamespacedName.Namespace})
+	}
+
+	serviceNamespacesNames = append(serviceNamespacesNames, clusterNamespacedName)
+
+	for _, serviceNamespacesName := range serviceNamespacesNames {
+		service := &corev1.Service{}
+
+		err = k8sClient.Get(ctx, serviceNamespacesName, service)
+		if err != nil {
+			return err
+		}
+
+		portSet := sets.NewInt32(ports...)
+
+		for _, p := range service.Spec.Ports {
+			if portSet.Has(p.Port) {
+				portSet.Delete(p.Port)
+			}
+		}
+
+		if portSet.Len() > 0 {
+			return fmt.Errorf("service %s port not configured correctly", serviceNamespacesName.Name)
+		}
```

Comment:
nit: why not just check and move the portSet outside of for loop
```
if !portSet.Has(p.Port) {
				return fmt.Errorf()
			}
```

Reply:
multiple ways to implement this anyway, here just didn't want to return error if any port is also available in port list other than the required ones. I know in test cases it will not happen.